static/app/views/starfish/queries/useSpanMetrics.tsx

```diff
@@ -37,14 +32,20 @@ export const useSpanMetrics = (
     Boolean(group) && Object.values(queryFilters).every(value => Boolean(value));
 
   // TODO: Add referrer
-  const result = useSpansQuery<SpanMetrics[]>({
+  const result = useSpansQuery({
     eventView,
     initialData: [],
     enabled,
     referrer,
   });
 
-  return {...result, data: result?.data?.[0] ?? {}, isEnabled: enabled};
+  const data = (result?.data?.[0] ?? {}) as Pick<MetricsResponse, T[number]>;
```

Comment:
I didn't have a chance to update the upstream typing, so I did `as` here to coerce it. It would be even better to type the whole hook so the code inside has to conform to the contract it declares

Reply:
`T[number]` sets the type of the returned data to the type of an elements inside `T`, which is the exact value of the `fields` argument